rdprail-shell/app-list.c

```diff
@@ -241,57 +247,84 @@ is_desktop_file(char *file)
 	return NULL;
 }
 
-static char *
-find_icon_file(char *name)
+static bool
+find_icon_file(struct app_entry *entry)
 {
+	struct desktop_shell *shell = entry->shell;
+	struct app_list_context *context = (struct app_list_context *)shell->app_list_context;
 	char buf[512];
 	int len;
+	char *icon_file = buf;
 
-	/* if name is absolute path and file presents, use as-is */ 
-	if (*name == '/') {
-		if (is_file_exist(name))
-			return strdup(name);
-		return NULL;
-	}
+	assert(entry->icon_name);
+	assert(entry->icon_file == NULL);
+	assert(entry->icon_retry_count < MAX_ICON_RETRY_COUNT);
 
-	/* TODO: follow icon search path desribed at "Icon Lookup" section at
+	/* if name is absolute path and file presents, use as-is */ 
+	if (*entry->icon_name == '/') {
+		if (is_file_exist(entry->icon_name)) {
+			icon_file = entry->icon_name;
+			goto Found;

```

Comment:
Maybe I'm confused, but if this path is taken are we leaking the original entry->icon_name string? icon_file will point to the original name, but then the original name will be overwritten in the Found section and I don't see us freeing it first.

Reply:
do you mean https://github.com/microsoft/weston-mirror/blob/52b4ce07ba089c750883a8b35c08d44d8cfd62a8/rdprail-shell/app-list.c#L304 ?